src/main/java/htsjdk/samtools/SamFiles.java

```diff
@@ -24,7 +25,7 @@
      * @return The index for the provided SAM, or null if one was not found.
      */
     public static File findIndex(final File samFile) {
-        Path path = findIndex(samFile.toPath());
+        final Path path = findIndex(IOUtil.toPath(samFile));
```

Comment:
It doesn't look like this will change the program behavior. This will cause `null` to be passed to `findIndex(Path)`, which will call `lookForIndex(Path)`, which will generate a NPE on line 69 below when it calls `samPath.getFileName()`.

If you do want to support `null` here, it would be easier to check `if (samFile == null)` directly IMO.